### PR TITLE
Remove and stop copying AsciiDoc files [skip ci]

### PR DESCRIPTION
**Pull request by mbrukman**
Remove CHANGELOG.asc and UPGRADE.asc from the repo (which are just copies of
files from the `docs` subdirectory), and delete instructions to update them in
RELEASING.md; also remove out-of-date information about `janusgraph-site` repo
which no longer exists from both release-related docs.

Partially addresses issue #615 and issue #729.

Should this PR target `0.2` branch or `master`?

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [] Have you written and/or updated unit tests to verify your changes?
- [] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [] Have you ensured that format looks appropriate for the output in which it is rendered?
- [x] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by robertdale** [comment]
I would prefer it targeted 0.2 otherwise it would have a slightly different release process. I think it's a worthy goal to keep them as close as possible.

**Comment by mbrukman** [comment]
@robertdale — rebased onto the `0.2` branch and updated the PR base branch.

File: RELEASING.md [hunk 1]
```diff
@@ -34,26 +34,14 @@ Release Steps
...
+build a set of docs for this release and copy them to the cloned
+`docs.janusgraph.org` repo which you will update later.
 
-Documentation pages:
+You may also need to update the following files in the main repo for any new
+or updated dependencies:
 
-* changelog.txt
-* versions.txt
-* upgrade.txt
```
Comment (robertdale) [inline RELEASING.md]: I couldn't find instructions elsewhere that remind us to update the above files under `docs/`.  I think this is an important step to be documented. How do you plan to handle?

Comment (mbrukman) [reply RELEASING.md]: Sorry, that was an extra deletion; I should have only deleted the next list of bulleted items (copying the same files to the top-level of the repo). Added back, cleaned up formatting.

**Pull request merged**